docs(common): Document NuGet API keys (#1787)
telerik/blazor-docs: installation/nuget.md

---
title: Telerik Private NuGet Feed
page_title: Telerik NuGet Feed

> description: How to add and use the Telerik private NuGet source.

Reviewer: Try adding more detailed descriptions, they must contain around 150 symbols.
```suggestion
description: Explore the different approaches of adding the Telerik private NuGet source to your system, and start using NuGet packages to install the Telerik Blazor components.
```

# Telerik Private NuGet Feed

This article explains how to add the private Telerik NuGet package source to your environment. You can use it to obtain the Telerik UI for Blazor components instead of [setting up a local NuGet feed]({%slug installation/zip%}#set-up-a-local-nuget-feed-in-visual-studio).

> There are several ways to setup the remote Telerik NuGet feed:

Reviewer: ```suggestion
You can set up the remote Telerik NuGet feed in the following ways:
```

* [Use Visual Studio](#use-visual-studio)
* [Use the .NET CLI](#use-the-net-cli)
* [Edit the Nuget.Config file](#edit-the-nugetconfig-file)

> >tip When working with the .NET CLI or editing the `NuGet.Config` manually, you can use your Telerik account credentials or a [NuGet API Key](#use-nuget-api-key). If you are logging at telerik.com through single sign on (SSO), use a [NuGet API Key](#use-nuget-api-key).

Reviewer: ```suggestion
>tip When working with the .NET CLI or editing the `NuGet.Config` manually, you can use your Telerik account credentials or a [NuGet API Key](#use-nuget-api-key). If you are logging in to telerik.com through single sign-on (SSO), use a [NuGet API Key](#use-nuget-api-key).
```

For NuGet-related issues, see [NuGet Feed Troubleshooting]({%slug troubleshooting-nuget%}).

For information on automated builds, CI and CD, see [CI, CD, Build Server]({%slug deployment-ci-cd-build-pc%}).

## Use Visual Studio

When adding NuGet sources in Visual Studio, the credentials are encrypted and stored outside the `NuGet.Config` file.

Refer to the [Microsoft documentation about using packages in Visual Studio](https://learn.microsoft.com/en-us/nuget/consume-packages/install-use-packages-visual-studio), or follow the steps below for Visual Studio on Windows.

1. Open Visual Studio and go to Tools > NuGet Package Manager > Package Manager Settings > Package Sources.

> 1. Click the `[+]` button at the top right hand side.

Reviewer: Buttons and other UI elements must be **bold**.
```suggestion
1. Click the **+** button at the top right-hand side.
```

1. Add the Telerik Feed URL `https://nuget.telerik.com/v3/index.json` and choose a Name for that package source (for example, "TelerikOnlineFeed").

1. Click OK.

1. Open a project that references a Telerik NuGet package. For example, generate one through our [New Project Wizard]({%slug getting-started-vs-integration-new-project%}).
    * Make sure to remove local `NuGet.Config` files from the solution that contain information about Telerik packages. 
    
1. Rebuild the solution.

1. A Windows prompt will ask for the Telerik feed credentials. Enter your Telerik email and password.

>     * Check "Remember my password".

Reviewer: ```suggestion
    * Check **Remember my password**.
```

>     * Check **Remember my password**.

Author: ```suggestion
    * Check the Remember My Password checkbox.
```
